raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -151,7 +155,10 @@ def try_new_route(state):
 
 
 def handle_block(state, state_change):
-    state.block_number = state_change.block_number
+    state.block_number = max(
+        state.block_number,
+        state_change.block_number,
+    )
```

Comment:
Right event, wrong place.

The mediated transfer tasks are really coded to move the protocol forward and have no notion of undoing events, there is no logic in-place to undo the events back to the fork point. I also think it's not a good idea to add code to handle reorgs in the mediated transfer tasks, as this would mix up concerns and definitely make the protocol code even harder to grasp.

The WAL OTOH is the perfect structure to unwind events, since it keeps all the history, we only need to get the newest snapshot *before* the reorg happened, and replay all the events up to the tip of the reorg. I would like to make a `StateManager` implementation that is aware of reorgs and can (re)create tasks using the above strategy.